cmd/demo/node_setup.go

```diff
@@ -0,0 +1,235 @@
+	if err != nil {
+		return nil, errors.WithMessage(err, "connecting to ethereum node")
+	}
+
+	n := &node{
+		log:     log.Get(),
+		onChain: acc,
+		wallet:  wallet,
+		dialer:  dialer,
+		cb:      echannel.NewContractBackend(ethereumBackend, wallet.Ks, &acc.Account),
+		peers:   make(map[string]*peer),
+	}
+
+	if err := n.setupContracts(); err != nil {
+		return nil, errors.WithMessage(err, "setting up contracts")
+	}
+	return n, n.setup()
+}
+
+// importAccount is a helper method to import secret keys until we have the ethereum wallet done.
+func importAccount(secret string) (*ewallet.Wallet, *ewallet.Account, error) {
+	ks := keystore.NewKeyStore(config.WalletPath, 2, 1)
+	sk, err := crypto.HexToECDSA(secret[2:])
+	if err != nil {
+		return nil, nil, errors.WithMessage(err, "decoding secret key")
+	}
+	var ethAcc accounts.Account
+	addr := crypto.PubkeyToAddress(sk.PublicKey)
+	if ethAcc, err = ks.Find(accounts.Account{Address: addr}); err != nil {
+		ethAcc, err = ks.ImportECDSA(sk, "")
+		if err != nil && errors.Cause(err).Error() != "account already exists" {
+			return nil, nil, errors.WithMessage(err, "importing secret key")
+		}
+	}
+
+	wallet, err := ewallet.NewWallet(ks, "")
+	if err != nil {
+		return nil, nil, errors.WithMessage(err, "creating wallet")
+	}
+
+	wAcc := ewallet.NewAccountFromEth(wallet, &ethAcc)
+	acc, err := wallet.Unlock(wAcc.Address())
+	return wallet, acc.(*ewallet.Account), err
+}
+
+func (n *node) PrintConfig() error {
+	fmt.Printf(
+		"Alias: %s\n"+
+			"Listening: %s:%d\n"+
+			"ETH RPC URL: %s\n"+
+			"Perun ID: %s\n"+
+			"OffChain: %s\n"+
+			"ETHAssetHolder: %s\n"+
+			"Adjudicator: %s\n"+
+			"", config.Alias, config.Node.IP, config.Node.Port, config.Chain.URL, n.onChain.Address().String(), n.offChain.Address().String(), n.assetAddr.String(), n.adjAddr.String())
+
+	fmt.Println("Known peers:")
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, 3, ' ', tabwriter.TabIndent)
+	for alias, peer := range config.Peers {
+		fmt.Fprintf(w, "%s\t%v\t%s:%d\n", alias, peer.PerunID, peer.Hostname, peer.Port)
+	}
+	return w.Flush()
+}
+
+// deployAdjudicator deploys the Adjudicator to the blockchain and returns its address
+// or an error.
+func deployAdjudicator(cb echannel.ContractBackend) (common.Address, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), config.Chain.TxTimeout)
+	defer cancel()
+	adjAddr, err := echannel.DeployAdjudicator(ctx, cb)
+	return adjAddr, errors.WithMessage(err, "deploying eth adjudicator")
+}
+
+// deployAsset deploys the Assetholder to the blockchain and returns its address
+// or an error. Needs an Adjudicator address as second argument.
+func deployAsset(cb echannel.ContractBackend, adjudicator common.Address) (common.Address, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), config.Chain.TxTimeout)
+	defer cancel()
+	asset, err := echannel.DeployETHAssetholder(ctx, cb, adjudicator)
+	return asset, errors.WithMessage(err, "deploying eth assetholder")
+}
```

Comment:
Please move below `setupContracts`. In general, it is easier to read if you order your functions etc from more to less abstract.